### DISCUSSION TITLE
C#: Allow silencing of assembly load exceptions

### DISCUSSION THREAD

**Discussion by impulsh** [citizenfx/fivem#2008] [Engineering Discussion]
Some libraries will try and load assemblies that might not exist, either to check for available features or use one of a few libraries it interfaces with (e.g an ORM that checks for multiple MySQL client libraries to use). This results in assembly load exceptions being logged to the console that aren't actually an issue.

Allowing silencing of the log message given [here](https://github.com/citizenfx/fivem/blob/9b9823fd1add1f833f8a3e116cb96ef78b0775ee/code/client/clrcore/InternalManager.cs#L266) would be ideal.

**Comment by thorium-cfx** [comment]
I'm quite hesitant on these requests, silencing such messages would leave debugging any of its symptoms harder and in the end we at Cfx end up with issue reports due to user errors.

Wondering if this is better addressed by developers themselves. Also I'll be turning this into a discussion as I see no issue.

**Reply by impulsh** [reply]
I think the ideal solution would be to simply rethrow the exception and let that be the "log message" for end users. I didn't suggest this originally as it would obviously cause compatibility issues with existing resources, but this might be a good change for the v2 runtime?

**Reply by thorium-cfx** [reply]
I'm not sure what you're suggesting or how that would help.

End-users would be the players and rethrowing them would result in the script-runtime needing to catch it or we'd be invalidating the whole resource/script.

**Reply by impulsh** [reply]
Whoops, you're right - I misunderstood how `AssemblyResolve` works. I'd still suggest that there be an "I know what I'm doing" opt-out for the log message in the resource manifest or similar.

I'm not sure how developers could address the issue since trying to load a type from a potentially non-existent assembly via something like `Type.GetType` is still a valid operation, and is done in some libraries that you can't modify if you're pulling straight from nuget.

**Reply by thorium-cfx** [reply]
Ah now I see what you're at, that'll require some testing. But you're free to try it out, including the throwing in there.
